From cbe63230ec9a53f2b4d30f1dcb22f7e8d2409663 Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Mon, 25 Jan 2021 20:32:09 +0200
Subject: [PATCH 48/78] s32gen1: fsl-qspi: Add pinmuxing for QSPI

Since pinctrl driver refactor in u-boot, the QSPI pinmuxing is not
applied by default from u-boot. Therefore, we must add pinmuxing from
the QSPI Linux driver.

Issue: ALB-6355
Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
---
 .../boot/dts/freescale/fsl-s32g274a-evb.dts   |  4 ++-
 .../boot/dts/freescale/fsl-s32g274a-rdb.dtsi  |  4 ++-
 .../boot/dts/freescale/fsl-s32g274a.dtsi      | 31 ++++++++++++++++-
 .../boot/dts/freescale/fsl-s32r45-evb.dts     | 33 ++++++++++++++++++-
 include/dt-bindings/pinctrl/s32g274-pinctrl.h | 16 ++++++---
 include/dt-bindings/pinctrl/s32r45-pinctrl.h  | 18 +++++++---
 6 files changed, 92 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
index 4a221716707d..d97018daecb3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 /*
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -51,6 +51,8 @@
 };
 
 &qspi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_qspi>;
 	status = "okay";
 
 	mx25uw51245g@0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
index 71c37edddb2c..0b6e7e637134 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb.dtsi
@@ -2,7 +2,7 @@
 /*
  * Device Tree file for S32G274ARDB Board.
  *
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  *
  */
 
@@ -612,6 +612,8 @@
 };
 
 &qspi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_qspi>;
 	status = "okay";
 
 	mx25uw51245g@0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
index bcdc037ae659..6594e6a3b919 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -648,6 +648,35 @@
 				S32_GEN1_PAD_PC_12_LLCE_CAN0_TX
 			>;
 		};
+
+		pinctrl0_qspi: qspigrp {
+				fsl,pins = <
+					S32_GEN1_PAD_PF13__QSPI_A_DQS_OUT
+					S32_GEN1_PAD_PF13__QSPI_A_DQS_IN
+					S32_GEN1_PAD_PG4__QSPI_A_CS0
+					S32_GEN1_PAD_PG5__QSPI_A_CS1
+					S32_GEN1_PAD_PF5__QSPI_A_DATA0_IN
+					S32_GEN1_PAD_PF5__QSPI_A_DATA0_OUT
+					S32_GEN1_PAD_PF6__QSPI_A_DATA1_IN
+					S32_GEN1_PAD_PF6__QSPI_A_DATA1_OUT
+					S32_GEN1_PAD_PF7__QSPI_A_DATA2_IN
+					S32_GEN1_PAD_PF7__QSPI_A_DATA2_OUT
+					S32_GEN1_PAD_PF8__QSPI_A_DATA3_IN
+					S32_GEN1_PAD_PF8__QSPI_A_DATA3_OUT
+					S32_GEN1_PAD_PF9__QSPI_A_DATA4_IN
+					S32_GEN1_PAD_PF9__QSPI_A_DATA4_OUT
+					S32_GEN1_PAD_PF10__QSPI_A_DATA5_IN
+					S32_GEN1_PAD_PF10__QSPI_A_DATA5_OUT
+					S32_GEN1_PAD_PF11__QSPI_A_DATA6_IN
+					S32_GEN1_PAD_PF11__QSPI_A_DATA6_OUT
+					S32_GEN1_PAD_PF12__QSPI_A_DATA7_IN
+					S32_GEN1_PAD_PF12__QSPI_A_DATA7_OUT
+					S32_GEN1_PAD_PG0__QSPI_A_SCK
+					S32_GEN1_PAD_PG1__QSPI_A_B_SCK
+					S32_GEN1_PAD_PG2__QSPI_A_SCK2
+					S32_GEN1_PAD_PG3__QSPI_A_B_SCK2
+				>;
+		};
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts b/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
index 99535899a84d..c3fd62ea1bd5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -170,6 +170,35 @@
 			>;
 		};
 
+		pinctrl0_qspi: qspigrp {
+				fsl,pins = <
+					S32_GEN1_PAD_PF13__QSPI_A_DQS_OUT
+					S32_GEN1_PAD_PF13__QSPI_A_DQS_IN
+					S32_GEN1_PAD_PG4__QSPI_A_CS0
+					S32_GEN1_PAD_PG5__QSPI_A_CS1
+					S32_GEN1_PAD_PF5__QSPI_A_DATA0_IN
+					S32_GEN1_PAD_PF5__QSPI_A_DATA0_OUT
+					S32_GEN1_PAD_PF6__QSPI_A_DATA1_IN
+					S32_GEN1_PAD_PF6__QSPI_A_DATA1_OUT
+					S32_GEN1_PAD_PF7__QSPI_A_DATA2_IN
+					S32_GEN1_PAD_PF7__QSPI_A_DATA2_OUT
+					S32_GEN1_PAD_PF8__QSPI_A_DATA3_IN
+					S32_GEN1_PAD_PF8__QSPI_A_DATA3_OUT
+					S32_GEN1_PAD_PF9__QSPI_A_DATA4_IN
+					S32_GEN1_PAD_PF9__QSPI_A_DATA4_OUT
+					S32_GEN1_PAD_PF10__QSPI_A_DATA5_IN
+					S32_GEN1_PAD_PF10__QSPI_A_DATA5_OUT
+					S32_GEN1_PAD_PF11__QSPI_A_DATA6_IN
+					S32_GEN1_PAD_PF11__QSPI_A_DATA6_OUT
+					S32_GEN1_PAD_PF12__QSPI_A_DATA7_IN
+					S32_GEN1_PAD_PF12__QSPI_A_DATA7_OUT
+					S32_GEN1_PAD_PG0__QSPI_A_SCK
+					S32_GEN1_PAD_PG1__QSPI_A_B_SCK
+					S32_GEN1_PAD_PG2__QSPI_A_SCK2
+					S32_GEN1_PAD_PG3__QSPI_A_B_SCK2
+				>;
+		};
+
 		pinctrl0_dspi1: dspi1grp {
 			fsl,pins = <
 				S32_GEN1_PAD_PB6__SPI1_SOUT_OUT
@@ -584,6 +613,8 @@
 };
 
 &qspi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl0_qspi>;
 	status = "okay";
 
 	mx25uw51245g@0 {
diff --git a/include/dt-bindings/pinctrl/s32g274-pinctrl.h b/include/dt-bindings/pinctrl/s32g274-pinctrl.h
index b2c57ee2f8b5..4f751795b0dd 100644
--- a/include/dt-bindings/pinctrl/s32g274-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32g274-pinctrl.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018-2020 NXP
+ * Copyright 2018-2021 NXP
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
@@ -276,7 +276,7 @@
 
 /* QSPI configuration */
 /* Note as-is DQS not enabled OR PAD_CTL_QSPI_A_DQS to enable */
-#define S32_GEN1_PAD_PF13__QSPI_A_DQS        548 PAD_CTL_SRC_SIG_SEL2
+#define S32_GEN1_PAD_PF13__QSPI_A_DQS_IN     548 PAD_CTL_SRC_SIG_SEL2
 #define S32_GEN1_PAD_PD4__QSPI_B_DQS         558 PAD_CTL_SRC_SIG_SEL2
 
 #define S32_GEN1_PAD_PF5__QSPI_A_DATA0_IN    540 PAD_CTL_SRC_SIG_SEL2
@@ -327,7 +327,7 @@
 #define S32_GEN1_PAD_PD5__QSPI_B_DATA7_IN    555 PAD_CTL_SRC_SIG_SEL2
 #define S32_GEN1_PAD_PD5__QSPI_B_DATA7_OUT   53 PAD_CTL_QSPI_B_DATA0_7
 
-#define S32_GEN1_PAD_PG5__QSPI_A_CS1         92  (PAD_CTL_QSPI_CLK_BASE | \
+#define S32_GEN1_PAD_PG5__QSPI_A_CS1         101  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_A_CS1_MUX)
 #define S32_GEN1_PAD_PD1__QSPI_B_CS1         93  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_B_CS1_MUX)
@@ -342,8 +342,14 @@
 #define S32_GEN1_PAD_PD6__QSPI_B_SCK         54  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_B_SCK_MUX)
 
-#define S32_GEN1_PAD_PG2__QSPI_CK2           98  (PAD_CTL_QSPI_CLK_BASE | \
-						  PAD_CTL_QSPI_CK2_MUX)
+#define S32_GEN1_PAD_PF13__QSPI_A_DQS_OUT    93  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG1__QSPI_A_B_SCK       97  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG2__QSPI_A_SCK2        98  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG3__QSPI_A_B_SCK2      99  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
 
 /* I2C configuration */
 #define S32_GEN1_PAD_PB0__I2C0_DATA_OUT      16  PAD_CTL_I2C0_MSCR_SDA_PB0
diff --git a/include/dt-bindings/pinctrl/s32r45-pinctrl.h b/include/dt-bindings/pinctrl/s32r45-pinctrl.h
index d98cbac6b80a..9b9a16a004be 100644
--- a/include/dt-bindings/pinctrl/s32r45-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32r45-pinctrl.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018-2020 NXP
+ * Copyright 2018-2021 NXP
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
@@ -247,7 +247,7 @@
 
 /* QSPI configuration */
 /* Note as-is DQS not enabled OR PAD_CTL_QSPI_A_DQS to enable */
-#define S32_GEN1_PAD_PF13__QSPI_A_DQS        548 PAD_CTL_SRC_SIG_SEL2
+#define S32_GEN1_PAD_PF13__QSPI_A_DQS_IN     548 PAD_CTL_SRC_SIG_SEL2
 #define S32_GEN1_PAD_PD4__QSPI_B_DQS         558 PAD_CTL_SRC_SIG_SEL2
 
 #define S32_GEN1_PAD_PF5__QSPI_A_DATA0_IN    540 PAD_CTL_SRC_SIG_SEL2
@@ -298,7 +298,7 @@
 #define S32_GEN1_PAD_PD5__QSPI_B_DATA7_IN    555 PAD_CTL_SRC_SIG_SEL2
 #define S32_GEN1_PAD_PD5__QSPI_B_DATA7_OUT   53 PAD_CTL_QSPI_B_DATA0_7
 
-#define S32_GEN1_PAD_PG5__QSPI_A_CS1         92  (PAD_CTL_QSPI_CLK_BASE | \
+#define S32_GEN1_PAD_PG5__QSPI_A_CS1         101  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_A_CS1_MUX)
 #define S32_GEN1_PAD_PD1__QSPI_B_CS1         93  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_B_CS1_MUX)
@@ -313,8 +313,16 @@
 #define S32_GEN1_PAD_PD6__QSPI_B_SCK         54  (PAD_CTL_QSPI_CLK_BASE | \
 						  PAD_CTL_QSPI_B_SCK_MUX)
 
-#define S32_GEN1_PAD_PG2__QSPI_CK2           98  (PAD_CTL_QSPI_CLK_BASE | \
-						  PAD_CTL_QSPI_CK2_MUX)
+#define S32_GEN1_PAD_PF13__QSPI_A_DQS_OUT    93  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG1__QSPI_A_B_SCK       97  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG2__QSPI_A_SCK2        98  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+#define S32_GEN1_PAD_PG3__QSPI_A_B_SCK2      99  (PAD_CTL_SRC_SIG_SEL1 | \
+						  PAD_CTL_QSPI_A_DQS)
+
+
 
 /* I2C configuration */
 #define S32_GEN1_PAD_PB0__I2C0_DATA_OUT      16  PAD_CTL_I2C0_MSCR_SDA
-- 
2.25.1

